pkg/cmd/issue/create/create.go

```diff
@@ -151,22 +151,22 @@ func createRun(opts *CreateOptions) (err error) {
 
 	tpl := shared.NewTemplateManager(httpClient, baseRepo, opts.RootDirOverride, !opts.HasRepoOverride, false)
 
+	var openURL string
+
 	if opts.WebMode {
-		openURL := ghrepo.GenerateRepoURL(baseRepo, "issues/new")
 		if opts.Title != "" || opts.Body != "" || tb.HasMetadata() {
-			if len(opts.Projects) > 0 {
-				var err error
-				tb.Projects, err = api.ProjectNamesToPaths(apiClient, baseRepo, tb.Projects)
-				if err != nil {
-					return fmt.Errorf("could not add to project: %w", err)
-				}
-			}
-			openURL, err = prShared.WithPrAndIssueQueryParams(openURL, tb)
+			openURL, err = generatePreviewURL(apiClient, baseRepo, tb)
 			if err != nil {
 				return
 			}
+			if !utils.ValidURL(openURL) {
+				return fmt.Errorf("Failed to create URL: maximum URL length exceeded")
```

Comment:
Due to our failure recovery mechanism in `issue/pr create`, new errors should first be assigned to `err` before they are returned.
```suggestion
				err = fmt.Errorf("Failed to create URL: maximum URL length exceeded")
				return
```

Reply:
Do you think `gh` should offer recover instructions for this error? If so, should I add more tests for this case?
I tried assigning errors to `err` but it in order to make it work I had to move the defer statement `preserveInput` right before this block.